### PR TITLE
Update docusaurus to v3

### PR DESCRIPTION
**Pull request by qwerty287**
@pat-s 

This would be my variant…

I'd prefer it as it does not changes the MD files.

### CODE REVIEW

**Comment by pat-s** [comment]
Looks way cleaner! Let's see if it builds and how it looks.

One thing I think we should change nevertheless are the "format" outputs in `cli.md` which are not formatted as code and contain c/p ascii characters.

**Comment by pat-s** [comment]
CI fails at the same error I encountered locally

> message: 'Hook useColorMode is called outside the <ColorModeProvider>. 

this seems (for now) unrelated to the plugins?

> I agree, but isn't this document auto-generated? [main/cmd/cli/docs.go#L26](https://github.com/woodpecker-ci/woodpecker/blob/main/cmd/cli/docs.go?rgh-link-date=2023-11-04T09%3A08%3A03Z#L26)

Ah, didn't know this. Makes sense though.
Then we need another approach (maybe some post-render hook) but as long as it's not blocking the general upgrade...

**Comment by pat-s** [comment]
Yeah, but I don't see how `useColorMode` and the `/api` relate. Until now, I thought of this being a general issue with the `useColorMode` definition in docusaurus v3 as described in the link of the error. But TBH I don't have enough react knowledge to really troubleshoot this.

**Comment by pat-s** [comment]
`redocusaurus` might take some time as the maintainer said he has no time. Anyhow, the error with the colorMode has been mentioned there before the 3.0 update and I wonder if it's actually related to it - or if we can fix it ourselves and maybe no issues exists with `redocusaurus` at all.

**Comment by anbraten** [comment]
I assume redocusaurus is using a component / context-hook to access the current color-mode, but that seems to be removed / renamed from the error.

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -139,6 +139,7 @@ module.exports = {
       prism: {
```
Comment (anbraten) [inline docs/docusaurus.config.js:139]: Could you change it to the .ts file format

Comment (anbraten) [reply docs/docusaurus.config.js:139]: https://docusaurus.io/docs/typescript-support#typing-config

Comment (anbraten) [reply docs/docusaurus.config.js:139]: thanks

**Review by anbraten — APPROVED** [review]
deployment & changes lgtm

**Pull request merged**